source/server/admin/profiling_handler.cc

```diff
@@ -93,5 +93,34 @@ Http::Code TcmallocProfilingHandler::handlerHeapDump(Http::ResponseHeaderMap&,
   return Http::Code::NotImplemented;
 }
 
+Http::Code TcmallocProfilingHandler::handlerAllocationProfiler(Http::ResponseHeaderMap&,
+                                                               Buffer::Instance& response,
+                                                               AdminStream& admin_stream) {
+  Http::Utility::QueryParamsMulti query_params = admin_stream.queryParams();
+  const auto enableVal = query_params.getFirstValue("enable");
+  if (query_params.data().size() != 1 || !enableVal.has_value() ||
+      (enableVal.value() != "y" && enableVal.value() != "n")) {
+    response.add("?enable=<y|n>\n");
```

Comment:
unit tests for this?

Reply:
https://github.com/envoyproxy/envoy/pull/36136/files#diff-f0d3e61ee48c80c5c04ce7f6db4bcf48e37803f31daf48b5f83846e19fe0af24R94 should cover it.

Reply:
That's one case right? There are 2 failure cases: start a profile when one had already been started, and stop a profile when one had not been started?

Reply:
Oh sorry I was not reading the testcase properly. Thanks! :)